AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConfigurationTests.swift

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSMobileClient
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreConfigurationTests: XCTestCase {
+
+    func testConfigureWithSameSchemaDoesNotDeleteDatabase() throws {
+        Amplify.reset()
+
+        let previousVersion = "previousVersion"
+        let saveSuccess = expectation(description: "Save was successful")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(version: previousVersion))
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure(AmplifyConfiguration(dataStore: nil))
+        } catch {
+            XCTFail("Failed to initialize Amplify with \(error)")
+        }
+
+        let post = Post(title: "title", content: "content", createdAt: .now())
+
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                saveSuccess.fulfill()
+            case .failure(let error):
+                XCTFail("Error saving post \(error)")
+            }
+        })
+
+        wait(for: [saveSuccess], timeout: TestCommonConstants.networkTimeout)
+
+        Amplify.reset()
+
+        let querySuccess = expectation(description: "query was successful")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(version: previousVersion))
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure(AmplifyConfiguration(dataStore: nil))
+        } catch {
+            XCTFail("Failed to initialize Amplify with \(error)")
+        }
+
+        // Query for the previously saved post. Data should be retrieved successfully which indicates that
+        // the database file was not deleted after re-configuring Amplify when using the same model registry version
+        Amplify.DataStore.query(Post.self, byId: post.id) { result in
+            switch result {
+            case .success(let postResult):
+                guard let queriedPost = postResult else {
+                    XCTFail("could not retrieve post across Amplify re-configure")
+                    return
+                }
+                XCTAssertEqual(queriedPost.title, "title")
+                XCTAssertEqual(queriedPost.content, "content")
+                XCTAssertEqual(queriedPost.createdAt, post.createdAt)
+                querySuccess.fulfill()
+            case .failure(let error):
+                XCTFail("Failed to query post, error: \(error)")
+            }
+        }
+
+        wait(for: [querySuccess], timeout: TestCommonConstants.networkTimeout)
+
+        Amplify.DataStore.clear(completion: { _ in })
```

Comment:
can be moved to tearDown

Reply:
What about 
`
func reset() {
    Amplify.reset()
}
`
and
`
func tearDown() {
    Amplify.DataStore.clear(completion: { _ in })
}
`

Reply:
Updated